src/topology/bounded_continuous_function.lean

```diff
@@ -429,5 +429,90 @@ def of_normed_group_discrete {α : Type u} {β : Type v}
   (f : α  → β) (C : ℝ) (H : ∀x, norm (f x) ≤ C) : α →ᵇ β :=
 of_normed_group f C H continuous_of_discrete_topology
 
+/- Formula for norm. -/
+lemma norm_eq (f : α →ᵇ β) :
+  ∥f∥ = Inf {C : ℝ | 0 ≤ C ∧ ∀ (x : α), ∥f x∥ ≤ C} :=
+begin
+  rw [← sub_zero f, ← dist_eq_norm, bounded_continuous_function.dist_eq],
+  simp,
+end
+
 end normed_group
+
+section normed_space
+/-! In this section, if `β` is a normed space, then we show that the space of bounded
+continuous functions from `α` to `β` inherits a normed space structure, by using
+pointwise operations and checking that they are compatible with the uniform distance. -/
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+variables [topological_space α] [normed_group β] [normed_space 𝕜 β]
+variables {f g : α →ᵇ β} {x : α} {C : ℝ}
+
+instance : has_scalar 𝕜 (α →ᵇ β) :=
+⟨λc, λf, ⟨λx, c • f.1 x,
+  continuous.smul continuous_const f.2.left,
+  begin
+    cases f.2.right with C hbound,
+    use ∥c∥ * C,
+    intros,
+    have hnneg : 0 ≤ ∥c∥ := norm_nonneg c,
+    specialize hbound x y,
+    rw dist_eq_norm at hbound ⊢,
+    calc ∥c • f x - c • f y∥ = ∥c • (f x - f y)∥ : by rw smul_sub c (f x) (f y)
+    ... = ∥c∥ * ∥f x - f y∥ : norm_smul c (f x - f y)
+    ... ≤ ∥c∥ * C : mul_le_mul_of_nonneg_left hbound hnneg
+  end⟩⟩
+
+instance : module 𝕜 (α →ᵇ β) :=
+  module.of_core $
+  { smul     := (•),
+    smul_add := λ c f g, ext $ λ x, smul_add c (f x) (g x),
+    add_smul := λ c₁ c₂ f, ext $ λ x, add_smul c₁ c₂ (f x),
+    mul_smul := λ c₁ c₂ f, ext $ λ x, mul_smul c₁ c₂ (f x),
+    one_smul := λ f, ext $ λ x, one_smul 𝕜 (f x) }
+
+instance : vector_space 𝕜 (α →ᵇ β) :=
+{ .. bounded_continuous_function.module }
+
+lemma bounded_continuous_sub_smul (c : 𝕜) (f : α →ᵇ β) : ∥c • f∥ ≤ ∥c∥ * ∥f∥ :=
```

Comment:
In `mathlib` the word `sub` in the name usually means `a - b` somewhere in the statement. For estimates we use something like `bounded_continuous_smul_le`.

Probably we need a version of the trick with `linear_map.mk_continuous` + `mk_continuous_norm_le` from `analysis/normed_space/operator_norm` to avoid proving the same estimates twice. Though it can wait till another PR.

Reply:
So concretely this name should probably become
```
bounded_continuous_function.norm_smul_le_mul
```
which seems like a moutful, but because `f` has type `bounded_continuous function ...` you would be able to apply this lemma as `f.norm_smul_le_mul` which is already better.
If you think this is the only short and reasonable statement involving `f`, norm and smul, then you could shorten the name to `f.norm_smul`. In other words
```
bounded_continuous_function.norm_smul
```

Reply:
@jcommelin No, it is not the only short and reasonable such statement.  This lemma (`∥c • f∥ ≤ ∥c∥ * ∥f∥`) is just a helper lemma for the lemma that follows (`∥c • f∥ = ∥c∥ * ∥f∥`), which fully supersedes it.  The second one is the lemma which is really deserving of the name  `f.norm_smul` (I will rename it accordingly).

The first one doesn't even really need a name, is there a way of making an "anonymous" lemma, or a lemma private to this file?

Reply:
Yes, you can make it `private` or `protected`. But we usually advise against it, because we've seen time and again that they turn out to be useful anyway. (Otherwise you will find people write `le_of_eq f.norm_smul` in other places.
So I suggest you call this `bounded_continuous_function.norm_smul_le`, and don't make it private.

Reply:
You can also wrap a section of code in
```
namespace bounded_continuous_function

variables (f : α →ᵇ β)

[code goes here]

end bounded_continuous_function
```
and then you don't need the `bounded_continuous_function.` prefix on the lemma names.

Reply:
```suggestion
private lemma norm_smul_le_mul (c : 𝕜) (f : α →ᵇ β) : ∥c • f∥ ≤ ∥c∥ * ∥f∥ :=
```
You can add `private` in front of `lemma` and then it will no longer be (easily) accessible from outside the `normed_space` section.

Everything is already inside the `bounded_continuous_function` namespace, so this one can just be named `norm_smul_le_mul` to get the same effect as Johan's suggestion. You can then use "dot" notation in the proof of the next lemma (see forthcoming suggestion below).